tests/unit/amo/test_helpers.js

```diff
@@ -0,0 +1,38 @@
+import React from 'react';
+import { ConnectedRouter } from 'connected-react-router';
+import { mount } from 'enzyme';
+import { createMemoryHistory } from 'history';
+import { Provider } from 'react-redux';
+import { Route } from 'react-router-dom';
+
+import createStore from 'amo/store';
+import { onLocationChanged } from 'tests/unit/amo/helpers';
+
+describe(__filename, () => {
+  describe('onLocationChanged', () => {
+    it('returns the same payload as the connected-react-router action', () => {
+      const history = createMemoryHistory();
+      const { store } = createStore({ history });
+      const dispatchSpy = sinon.spy(store, 'dispatch');
+
+      mount(
+        <Provider store={store}>
+          <ConnectedRouter history={history}>
+            <Route path="/" render={() => <div>Welcome</div>} />
+          </ConnectedRouter>
+        </Provider>,
+      );
+
+      dispatchSpy.resetHistory();
+
+      const pathname = '/foo';
+      history.push(pathname);
+
+      sinon.assert.calledWith(
+        dispatchSpy,
+        sinon.match(onLocationChanged({ pathname })),
```

Comment:
I think this should not use `sinon.match()` because we *do* want to know if there are more or less props in the payload. Was there a dynamic value giving you trouble? If so, you could create a custom sinon matcher.

Reply:
Okay. Here's an example of a custom matcher:

https://github.com/mozilla/addons-frontend/blob/6dcad19b0dc9fb21ed6b42cfe351fc0ec803ede6/tests/unit/helpers.js#L845-L867

Reply:
I see. It seems a lot of code to me, for a matcher that won't be reused.

The use of `sinon.match` here catches the following bugs:

- `action` set to another constant in the upstream lib
- `location` not set or with a different shape (i.e. `pathname` not present anymore)

That's why I thought that would be enough. Let me come back with something stricter yet still simple.